yaptide/celery/utils/progress/fluka_monitor.py

```diff
@@ -0,0 +1,129 @@
+from dataclasses import dataclass
+from datetime import datetime, timezone
+import logging
+import re
+import threading
+from typing import Iterator, Optional, Tuple
+from yaptide.batch.watcher import TIMEOUT_MATCH
+
+from yaptide.celery.utils.requests import send_task_update
+from yaptide.utils.enums import EntityState
+
+# templates for regex matching output from `<simulation>_<no>.out` file
+S_OK_OUT_INIT = "Total time used for initialization:"
+S_OK_OUT_START = "1NUMBER OF BEAM"
+S_OK_OUT_IN_PROGRESS = " NEXT SEEDS:"
+S_OK_OUT_COLLECTED = " All cases handled by Feeder"
+S_OK_OUT_FIN_PATTERN = re.compile(r"^ \* ======(?:( )*?)End of FLUKA [\w\-.]* run (?:( )*?) ====== \*")
+
+logger = logging.getLogger(__name__)
+
+
+def parse_progress_remaining_line(line: str) -> Optional[Tuple[int, int]]:
```

Comment:
```suggestion
def parse_progress_remaining_line(line: str) -> Optional[tuple[int, int]]:
```

I think we can stick to python > 3.9